src/handler/target.rs

```diff
@@ -486,11 +488,11 @@ impl Target {
 #[derive(Debug, Clone)]
 pub struct TargetConfig {
     pub ignore_https_errors: bool,
-    pub viewport: Viewport,
+    pub viewport: Option<Viewport>,
 }
 
 impl TargetConfig {
-    pub fn new(ignore_https_errors: bool, viewport: Viewport) -> Self {
+    pub fn new(ignore_https_errors: bool, viewport: Option<Viewport>) -> Self {
```

Comment:
Since we use the `new` function only internally, this is fine, otherwise I'd prefer a separate `with_viewport` function

Reply:
I'll keep that in mind.